merlin/systems/dag/ops/hugectr.py

```diff
@@ -0,0 +1,300 @@
+        version : int, optional
+            The version of the model, by default 1
+
+        Returns
+        -------
+        config
+            Dictionary representation of config file in memory.
+        """
+        node_name = f"{node_id}_{self.export_name}" if node_id is not None else self.export_name
+        node_export_path = pathlib.Path(path) / node_name
+        node_export_path.mkdir(exist_ok=True)
+        model_name = node_name
+        hugectr_model_path = pathlib.Path(node_export_path) / str(version)
+        hugectr_model_path.mkdir(exist_ok=True)
+
+        network_file = os.path.join(hugectr_model_path, f"{model_name}.json")
+
+        self.model.graph_to_json(graph_config_file=network_file)
+        self.model.save_params_to_files(str(hugectr_model_path) + "/")
+        model_json = json.loads(open(network_file, "r").read())
+        dense_pattern = "*_dense_*.model"
+        dense_path = [
+            os.path.join(hugectr_model_path, path.name)
+            for path in hugectr_model_path.glob(dense_pattern)
+            if "opt" not in path.name
+        ][0]
+        sparse_pattern = "*_sparse_*.model"
+        sparse_paths = [
+            os.path.join(hugectr_model_path, path.name)
+            for path in hugectr_model_path.glob(sparse_pattern)
+            if "opt" not in path.name
+        ]
+
+        config_dict = dict()
+        config_dict["supportlonglong"] = True
+
+        data_layer = model_json["layers"][0]
+        sparse_layers = [
+            layer
+            for layer in model_json["layers"]
+            if layer["type"] == "DistributedSlotSparseEmbeddingHash"
+        ]
+        num_cat_columns = sum(x["slot_num"] for x in data_layer["sparse"])
+        vec_size = [x["sparse_embedding_hparam"]["embedding_vec_size"] for x in sparse_layers]
+
+        model = dict()
+        model["model"] = model_name
+        model["slot_num"] = num_cat_columns
+        model["sparse_files"] = sparse_paths
+        model["dense_file"] = dense_path
+        model["maxnum_des_feature_per_sample"] = data_layer["dense"]["dense_dim"]
+        model["network_file"] = network_file
+        model["num_of_worker_buffer_in_pool"] = 4
+        model["num_of_refresher_buffer_in_pool"] = 1
+        model["deployed_device_list"] = self.device_list
+        model["max_batch_size"] = self.max_batch_size
+        model["default_value_for_each_table"] = [0.0] * len(sparse_layers)
+        model["hit_rate_threshold"] = 0.9
+        model["gpucacheper"] = self.hugectr_params["gpucacheper"]
+        model["gpucache"] = True
+        model["cache_refresh_percentage_per_iteration"] = 0.2
+        model["maxnum_catfeature_query_per_table_per_sample"] = [
+            len(x["sparse_embedding_hparam"]["slot_size_array"]) for x in sparse_layers
+        ]
+        model["embedding_vecsize_per_table"] = vec_size
+        model["embedding_table_names"] = [x["top"] for x in sparse_layers]
```

Comment:
Wonder if might be worthwhile to extract a helper function to construct this dictionary